tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -88,32 +88,33 @@ def delete_and_create_osd_node(osd_node_name):
         f"'{config.ENV_DATA['deployment_type']}' is not valid, "
         f"results of this test run are all invalid."
     )
-    # TODO: refactor this so that AWS is not a "special" platform
-    if config.ENV_DATA["platform"].lower() == constants.AWS_PLATFORM:
-        if config.ENV_DATA["deployment_type"] == "ipi":
-            new_node_name = node.delete_and_create_osd_node_ipi(osd_node_name)
 
-        elif config.ENV_DATA["deployment_type"] == "upi":
-            new_node_name = node.delete_and_create_osd_node_aws_upi(osd_node_name)
-        else:
-            log.error(msg_invalid)
-            pytest.fail(msg_invalid)
-    elif config.ENV_DATA["platform"].lower() in constants.CLOUD_PLATFORMS:
-        if config.ENV_DATA["deployment_type"] == "ipi":
-            new_node_name = node.delete_and_create_osd_node_ipi(osd_node_name)
-        else:
-            log.error(msg_invalid)
-            pytest.fail(msg_invalid)
-    elif config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM:
+    if config.ENV_DATA["deployment_type"] == "ipi":
         if is_lso_cluster():
-            new_node_name = node.delete_and_create_osd_node_vsphere_upi_lso(
-                osd_node_name, use_existing_node=False
+            # TODO: Implement functionality for Internal-Attached devices mode
+            # once ocs-ci issue #4545 is resolved
+            # https://github.com/red-hat-storage/ocs-ci/issues/4545
+            raise NotImplementedError(
```

Comment:
shouldn't we skip? throwing an exception means this test will fail

Reply:
Right.
I've modified the code. Added skip instead of raising an exception.